101-machine-learning-create/azuredeploy.json

```diff
@@ -2,257 +2,641 @@
+      }
+    },
+    "cmk_keyvault":{
+      "type": "string",
+      "defaultValue": "",
+      "metadata": {
+        "description": "Specifies the customer managed keyVault arm id."
+      }
+    },
+    "resource_cmk_uri":{
+      "type": "string",
+      "defaultValue": "",
+      "metadata": {
+        "description": "Specifies if the customer managed keyvault key uri."
+      }
+    },
+    "privateEndpointType": {
+      "type": "string",
+      "defaultValue": "none",
+      "allowedValues": [
+        "AutoApproval",
+        "ManualApproval",
+        "none"
+      ]
+    }
+
+  },
+  "variables": {
+    "storageAccountName": "[if(empty(parameters('storageAccountName')), concat('sa',uniqueString(resourceGroup().id, parameters('location'))), parameters('storageAccountName'))]",
+    "keyVaultName": "[if(empty(parameters('keyVaultName')), concat('kv',uniqueString(resourceGroup().id, parameters('location'))), parameters('keyVaultName'))]",
+    "applicationInsightsName": "[if(empty(parameters('applicationInsightsName')), concat('ai',uniqueString(resourceGroup().id, parameters('location'))), parameters('applicationInsightsName'))]",
+    "containerRegistryName": "[if(empty(parameters('containerRegistryName')), concat('cr',uniqueString(resourceGroup().id, parameters('location'))), parameters('containerRegistryName'))]",
+    "vnetName": "[if(empty(parameters('vnetName')), concat('vn',uniqueString(resourceGroup().id, parameters('location'))), parameters('vnetName'))]",
+    "subnetName": "[if(empty(parameters('subnetName')), concat('sn',uniqueString(resourceGroup().id, parameters('location'))), parameters('subnetName'))]",
+    "tenantId": "[subscription().tenantId]",
+    "storageAccount": "[resourceId(parameters('storageAccountResourceGroupName'), 'Microsoft.Storage/storageAccounts', variables('storageAccountName'))]",
+    "keyVault": "[resourceId(parameters('keyVaultResourceGroupName'), 'Microsoft.KeyVault/vaults', variables('keyVaultName'))]",
+    "containerRegistry": "[resourceId(parameters('containerRegistryResourceGroupName'), 'Microsoft.ContainerRegistry/registries', variables('containerRegistryName'))]",
+    "applicationInsights": "[resourceId(parameters('applicationInsightsResourceGroupName'), 'Microsoft.Insights/components', variables('applicationInsightsName'))]",
+    "vnet": "[resourceId(parameters('vnetResourceGroupName'), 'Microsoft.Network/virtualNetworks', variables('vnetName'))]",
+    "subnet": "[resourceId(parameters('vnetResourceGroupName'), 'Microsoft.Network/virtualNetworks/subnets', variables('vnetName'), variables('subnetName'))]",
+    "privateDnsGuid": "[guid(resourceGroup().id, deployment().name)]",
+    "locationsPEAvailable": [
+      "southcentralus",
+      "eastus",
+      "westus2"
+    ],
+    "isLocationValid": "[or(equals(parameters('privateEndpointType'), 'none'), contains(variables('locationsPEAvailable'), parameters('location')))]",
```

Comment:
remove this variable (you'll need to remove the pattern that uses below it as well)

Reply:
@bmoore-msft For this one. I need to limit the location to the selected locations only if private endpoint is not none. Any idea to handle this if not using this way?